### PR TITLE
feat: modularize drawer components for setup page

### LINKED ISSUE #300: Refactor drawer components to be reusable in the setup
## Description

The current drawer components can only be used on pages outside of the setup, which defeats the purpose of reusability. In this task, I will refactor all the drawer components by separating the logic and UI parts to make them reusable.

## Criteria

- [ ] Refactor all drawer components by separating the logic and UI parts

### PR DESCRIPTION
**Pull request by akiomatic**
## Overview

I've made the following drawer components in the changes reusable on the setup page by decoupling the onSubmit logic and UI parts. In order to proceed with this refactoring, I've created a `CommonDrawer` component to create a drawer with a form without having to write the boilerplate code.

## Changes

- Added `CommonDrawer` component to structure the base of a drawer component

- Decoupled logic and UI of the following components and modularized in separate files (these files end with `Content.tsx`)
  - TenantInvitationDrawer
  - TaskCreationDrawer
  - CategoryCreationDrawer
  - NameEditionDrawer
  - DeleteConfirmationDrawer
  - TaskDeletionDrawer

- After that, added the following components for the setup page
  - SetupTenantInvitationDrawer
  - SetupTaskCreationDrawer
  - SetupCategoryCreationDrawer
  - SetupNameEditionDrawer
  - SetupDeleteConfirmationDrawer
  - SetupTaskDeletionDrawer

- No longer need to pass `formControls` to children components, now utilizes `<FormProvider>` and `useFormContext`
  - https://react-hook-form.com/docs/formprovider

- Changed the `mode` of `useForm` to `onBlur` as per our decision before

- In `TaskCreationDrawer` component, made it utilize `isDirty` state from `react-hook-form` to determine if the landlord actually modified any value in the drawer
  - https://react-hook-form.com/docs/useform/formstate

## Review points

Please confirm that all the drawers work as before.

## Notes

I'm sorry but I haven't added comments on the components introduced in this PR. I'll comment on them later in another PR.

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/drawers/names/NameEditionDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+ * />
+ *
+ * // To edit the category name
+ * <NameEditionDrawer
+ *  name={'sample name'}
+ *  open={open}
+ *  setOpen={setOpen}
+ *  type={'category'}
+ * />
+ */
```
Comment (Akikaze1119) [inline src/components/ui/drawers/names/NameEditionDrawer.tsx]: If possible, could you also add the `id` prop to the @param and @example?

**Review by Akikaze1119 — COMMENTED** [review]
Thank you Akio!
Could you check my comment? It's a very minor detail...

**Review by yuki-92c — APPROVED** [review]
LGTM. Thank you!

File: src/components/ui/drawers/names/NameEditionDrawer.tsx (hunk 1 above)
Comment (akiomatic) [reply src/components/ui/drawers/names/NameEditionDrawer.tsx]: @Akikaze1119 Hi. I've just added it!

**Review by Akikaze1119 — APPROVED** [review]
Thank you!! LGTM :)

**Pull request merged**